assets/js/modules/reader-revenue-manager/datastore/settings.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * `modules/reader-revenue-manager` data store: settings actions.
+ *
+ * Site Kit by Google, Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+
+/**
+ * Internal dependencies
+ */
+import { MODULES_READER_REVENUE_MANAGER } from './constants';
+import {
+	INVARIANT_DOING_SUBMIT_CHANGES,
+	INVARIANT_SETTINGS_NOT_CHANGED,
+} from '../../../googlesitekit/data/create-settings-store';
+import { createStrictSelect } from '../../../googlesitekit/data/utils';
+import { isValidPublicationID } from '../utils/validation';
+
+// Invariant error messages.
+export const INVARIANT_INVALID_PUBLICATION_ID =
+	'a valid publicationID is required';
+
+export function validateCanSubmitChanges( select ) {
```

Comment:
According to the IB this function should check the following, but it seems to have been missed:

> - Publication ID belongs to one of the publication IDs that we received from `getPublications` selector.

Reply:
The IB seems to be incorrect. I checked with @nfmohit, who confirmed that:
>Your version is correct. It is not necessary to validate the publication ID with the response of getPublications. We do not do that in case of any other modules as far as I know. Thank you for checking!

Reply:
Cool, thanks for checking and clarifying!